ta/pkcs11/src/processing_symm.c

```diff
@@ -712,3 +796,70 @@ enum pkcs11_rc step_symm_operation(struct pkcs11_session *session,
 
 	return rc;
 }
+
+enum pkcs11_rc derive_key_by_symm_enc(struct pkcs11_session *session,
+				      struct obj_attrs **head)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	struct active_processing *proc = session->processing;
+	struct input_data_ref *input = proc->extra_ctx;
+	void *in_buf = NULL;
+	void *out_buf = NULL;
+	uint32_t out_size = 0;
+	uint32_t in_size = 0;
+	uint32_t size = sizeof(uint32_t);
+	uint32_t key_length = 0;
+
+	if (!proc->extra_ctx)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	in_buf = input->data;
+	in_size = input->size;
+
+	out_size = in_size;
+	out_buf = TEE_Malloc(out_size, 0);
```

Comment:
maybe testing `out_size` and return `ARGUMENTS_BAD`?

Reply:
Yes, I will add the check